packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -257,7 +257,8 @@ export class Alarm extends AlarmBase {
     return dispatchMetric(metric, {
       withStat(stat, conf) {
         self.validateMetricStat(stat, metric);
-        if (conf.renderingProperties?.label == undefined) {
+        if (conf.renderingProperties?.label == undefined &&
```

Comment:
💅 for readability, it might make sense to assign this expression to a variable to indicate what and why this is happening.

```ts
const canRenderAsLegacyMetric = /* ...expression... */;

// Do this to disturb existing templates as little as possible
if (canRenderAsLegacyMetric) {
  return /* ... */;
}
```